pkg/auth/handler/webapp/csrf_error_instruction.go

```diff
@@ -0,0 +1,67 @@
+package webapp
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/authgear/authgear-server/pkg/auth/handler/webapp/viewmodels"
+	"github.com/authgear/authgear-server/pkg/auth/webapp"
+	"github.com/authgear/authgear-server/pkg/util/httproute"
+)
+
+func ConfigureCSRFErrorInstructionRoute(route httproute.Route) httproute.Route {
+	return route.
+		WithMethods("OPTIONS", "GET").
+		WithPathPattern("/csrf/error/instruction")
+}
+
+type CSRFErrorInstructionHandler struct {
+	Controller    *AuthflowController
+	BaseViewModel *viewmodels.BaseViewModeler
+	Renderer      Renderer
+}
+
+func (h *CSRFErrorInstructionHandler) GetData(w http.ResponseWriter, r *http.Request, s *webapp.Session, screen *webapp.AuthflowScreenWithFlowResponse) (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+
+	baseViewModel := h.BaseViewModel.ViewModelForAuthFlow(r, w)
+	viewmodels.Embed(data, baseViewModel)
+
+	userAgent := r.UserAgent()
+	device := h.getDevice(userAgent)
+	data["Device"] = device
+
+	return data, nil
+}
+
+func (h *CSRFErrorInstructionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	var handlers AuthflowControllerHandlers
+	handlers.Get(func(s *webapp.Session, screen *webapp.AuthflowScreenWithFlowResponse) error {
+		data, err := h.GetData(w, r, s, screen)
+		if err != nil {
+			return err
+		}
+
+		h.Renderer.RenderHTML(w, r, TemplateCsrfErrorInstructionHTML, data)
+		return nil
+	})
+	h.Controller.HandleWithoutFlow(w, r, &handlers)
+}
+
+func (h *CSRFErrorInstructionHandler) getDevice(userAgent string) (device string) {
+	device = "default"
+	if strings.Contains(userAgent, "Chrome") {
+		device = "ChromeDesktop"
+	}
+	if strings.Contains(userAgent, "Android") {
+		device = "Android"
+	}
+	if strings.Contains(userAgent, "iPhone") || strings.Contains(userAgent, "iPad") {
+		device = "iOS"
+	}
+	if strings.Contains(userAgent, "SM") {
+		device = "Samsung"
+	}
```

Comment:
You add a function in `pkg/api/model/user_agent.go` to do this logic.

You are going to use `uaParser` to parse the user agent string, you will get a `*uaparser.Client`. And then you use the parsed fields to determine `Chrome`, `iOS`, `ChromeAndroid`, `Samsung`

```
// The name is borrowed from https://github.com/browserslist/browserslist
type RecognizedMobileDevice string

const (
  // iOS is the Safari browser on iOS.
  RecognizedMobileDeviceIOS = "iOS"
  // ChromeAndroid is the Chrome browser on Android. It is NOT the Android webview.
  RecognizedMobileDeviceChromeAndroid = "ChromeAndroid"
  // Chrome is the Chrome browser on desktop.
  RecognizedMobileDeviceChrome = "Chrome"
  // Samsung is the Sumsung browser on a Samsung Android device.
  RecognizedMobileDeviceSamsung = "Samsung"
)

function GetRecognizedMobileDevice(userAgentString string) (RecognizedMobileDevice, bool) {
  // You return "", false as a fallback case. That's why I say you do not write (eq .Device "Default")
}
```